### PR TITLE
Revise ingress controller list

### PR DESCRIPTION
**Pull request by sftim**
Update https://k8s.io/docs/concepts/services-networking/ingress-controllers/ [[preview](https://deploy-preview-25026--kubernetes-io-master-staging.netlify.app/docs/concepts/services-networking/ingress-controllers/)]:

- note that AWS load balancer integration is now in-project
- sort alphabetically
- link to documentation for the product
- revise wording to always describe the ingress controller rather than
  the service / product that the ingress controller integrates with

/sig network

### CODE REVIEW

File: content/en/docs/concepts/services-networking/ingress-controllers.md [hunk 1]
```diff
@@ -24,31 +23,31 @@ Kubernetes as a project currently supports and maintains [GCE](https://git.k8s.i
...
+  controller.
+* The [Citrix ingress controller](https://github.com/citrix/citrix-k8s-ingress-controller#readme) works with
+  Citrix Application Delivery Controller.
+* [Contour](https://projectcontour.io/) is an [Envoy](https://www.envoyproxy.io/) based ingress controller.
+* F5 BIG-IP [Container Ingress Services for Kubernetes](https://clouddocs.f5.com/containers/latest/userguide/kubernetes/)
+  lets you use an Ingress to configure F5 BIG-IP virtual servers.
+* [Gloo](https://gloo.solo.io) is an open-source ingress controller based on [Envoy](https://www.envoyproxy.io),
+  which offers API gateway functionality.
+* [HAProxy Ingress](https://haproxy-ingress.github.io/) is an ingress controller for
+  [HAProxy](http://www.haproxy.org/#desc).
```
Comment (kbhawkey) [inline content/en/docs/concepts/services-networking/ingress-controllers.md:37]: @sftim, Could you list the two (?) `HAProxy` ingress controllers together?

Comment (sftim) [reply content/en/docs/concepts/services-networking/ingress-controllers.md:37]: There are three; Voyager is also an ingress controller for HAProxy. Two have quite similar names but are different projects.

**Comment by sftim** [comment]
PR #24777 covers just the AWS cloud provider controller rename.

File: content/en/docs/concepts/services-networking/ingress-controllers.md [hunk 2]
```diff
@@ -24,31 +23,31 @@ Kubernetes as a project currently supports and maintains [GCE](https://git.k8s.i
...
-  [Kong Ingress Controller for Kubernetes](https://github.com/Kong/kubernetes-ingress-controller).
-* [NGINX, Inc.](https://www.nginx.com/) offers support and maintenance for the
-  [NGINX Ingress Controller for Kubernetes](https://www.nginx.com/products/nginx/kubernetes-ingress-controller).
-* [Skipper](https://opensource.zalando.com/skipper/kubernetes/ingress-controller/) HTTP router and reverse proxy for service composition, including use cases like Kubernetes Ingress, designed as a library to build your custom proxy
-* [Traefik](https://github.com/traefik/traefik) is a fully featured ingress controller
-  ([Let's Encrypt](https://letsencrypt.org), secrets, http2, websocket), and it also comes with commercial
-  support by [Traefik Labs](https://traefik.io).
+* [AKS Application Gateway Ingress Controller](https://azure.github.io/application-gateway-kubernetes-ingress/) is an ingress controller that configures the [Azure Application Gateway](https://docs.microsoft.com/azure/application-gateway/overview).
+* [Ambassador](https://www.getambassador.io/) API Gateway is an [Envoy](https://www.envoyproxy.io)-based ingress
+  controller.
```
Comment (kbhawkey) [inline content/en/docs/concepts/services-networking/ingress-controllers.md:28]: Is the AWS Load Balancer missing?

Comment (sftim) [reply content/en/docs/concepts/services-networking/ingress-controllers.md:28]: The rendered page states:
> Kubernetes as a project supports and maintains [AWS](https://github.com/kubernetes-sigs/aws-load-balancer-controller#readme), [GCE](https://git.k8s.io/ingress-gce/README.md#readme), and [nginx](https://git.k8s.io/ingress-nginx/README.md#readme) ingress controllers.

If this is confusing people we could have a follow-up PR that turns the list into a table, with an “officially supported” column that is checked for AWS, GCE and the in-project nginx controller. I would do that as a follow-up though.

File: content/en/docs/concepts/services-networking/ingress-controllers.md (hunk 1 above)
Comment (kbhawkey) [reply content/en/docs/concepts/services-networking/ingress-controllers.md:37]: OK

File: content/en/docs/concepts/services-networking/ingress-controllers.md (hunk 2 above)
Comment (kbhawkey) [reply content/en/docs/concepts/services-networking/ingress-controllers.md:28]: Ah. You have removed this from the list entirely. I thought you originally had AWS in the list as well in the
"support and maintain" statement.

**Pull request merged**